packages/web/src/components/duplicate-add-confirmation-modal/DuplicateAddConfirmationModal.tsx

```diff
@@ -0,0 +1,120 @@
+import { useCallback, useContext } from 'react'
+
+import {
+  accountSelectors,
+  cacheCollectionsActions,
+  cacheCollectionsSelectors,
+  duplicateAddConfirmationModalUISelectors,
+  fillString
+} from '@audius/common'
+import {
+  Button,
+  ButtonType,
+  Modal,
+  ModalContent,
+  ModalContentText,
+  ModalFooter,
+  ModalHeader,
+  ModalTitle
+} from '@audius/stems'
+import { useDispatch } from 'react-redux'
+
+import { useModalState } from 'common/hooks/useModalState'
+import { useSelector } from 'common/hooks/useSelector'
+import { ToastContext } from 'components/toast/ToastContext'
+import ToastLinkContent from 'components/toast/mobile/ToastLinkContent'
+import { playlistPage } from 'utils/route'
+
+import styles from './DuplicateAddConfirmationModal.module.css'
+
+const { addTrackToPlaylist } = cacheCollectionsActions
+const { getCollection } = cacheCollectionsSelectors
+const { getPlaylistId, getTrackId } = duplicateAddConfirmationModalUISelectors
+const { getAccountUser } = accountSelectors
+
+const messages = {
+  title: 'Already Added',
+  description: 'This is already in your%0 playlist',
+  cancel: "Don't Add",
+  add: 'Add Anyway',
+  addedToast: 'Added To Playlist!',
+  view: 'View'
+}
+
+export const DuplicateAddConfirmationModal = () => {
+  const dispatch = useDispatch()
+  const { toast } = useContext(ToastContext)
+  const playlistId = useSelector(getPlaylistId)
+  const trackId = useSelector(getTrackId)
+  const playlist = useSelector((state) =>
+    getCollection(state, { id: playlistId })
+  )
+  const account = useSelector(getAccountUser)
+  const [isOpen, setIsOpen] = useModalState('DuplicateAddConfirmation')
+
+  const onClose = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  const handleAdd = useCallback(() => {
+    if (trackId && playlistId) {
+      dispatch(addTrackToPlaylist(trackId, playlistId))
+      if (account) {
+        toast(
+          <ToastLinkContent
```

Comment:
i think now you can just do `toast({content: messages.addedToast, link: playlistPage(...), linkText: messages.addedToast})`